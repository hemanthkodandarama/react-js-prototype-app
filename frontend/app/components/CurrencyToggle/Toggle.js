/**
 *
 * LocaleToggle
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import Arrow from 'components/Arrow';
import Select from 'components/Select';
import ToggleOption from '../ToggleOption';
import SelectWrapper from './SelectWrapper';

function Toggle(props) {
  let content = <option>--</option>;

  // If we have items, render them
  if (props.values) {
    content = props.values.map(value => (
      <ToggleOption key={value} value={value} message={props.messages[value]} />
    ));
  }

  return (
    <SelectWrapper>
      <Select value={props.value} onChange={props.onToggle}>
        {content}
      </Select>
      <Arrow />
    </SelectWrapper>
  );
}

Toggle.propTypes = {
  onToggle: PropTypes.func,
  values: PropTypes.array,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  messages: PropTypes.object,
};

export default Toggle;
