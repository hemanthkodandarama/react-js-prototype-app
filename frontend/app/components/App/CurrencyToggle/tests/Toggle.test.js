import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider, defineMessages } from 'react-intl';

import Toggle from '../Toggle';

describe('<Toggle />', () => {
  it('should contain default text', () => {
    const defaultEnMessage = 'someContent';
    const defaultDeMessage = 'someOtherContent';
    const messages = defineMessages({
      en: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultEnMessage,
      },
      de: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultDeMessage,
      },
    });
    const { container } = render(
      <IntlProvider locale="en">
        <Toggle values={['en', 'de']} messages={messages} />
      </IntlProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
