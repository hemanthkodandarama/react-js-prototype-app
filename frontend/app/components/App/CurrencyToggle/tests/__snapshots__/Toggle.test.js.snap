// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should contain default text 1`] = `
<div
  class="SelectWrapper-sc-1elffer-0 fZdpiz"
>
  <select
    class="Select-fhpbt0-0 cgpFOs"
  >
    <option
      value="en"
    >
      someContent
    </option>
    <option
      value="de"
    >
      someOtherContent
    </option>
  </select>
  <div
    class="Arrow-sc-172g500-0 jcSFyu"
  />
</div>
`;
